BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -298,3 +311,27 @@ extension CryptoStore: BraveWalletKeyringServiceObserver {
   public func selectedAccountChanged(_ coinType: BraveWallet.CoinType) {
   }
 }
+
+extension CryptoStore: BraveWalletBraveWalletServiceObserver {
+  public func onActiveOriginChanged(_ originInfo: BraveWallet.OriginInfo) {
+  }
+  
+  public func onDefaultWalletChanged(_ wallet: BraveWallet.DefaultWallet) {
+  }
+  
+  public func onDefaultBaseCurrencyChanged(_ currency: String) {
+    currencyFormatter.currencyCode = currency
+    portfolioStore.update()
+    assetDetailStore?.update()
+    accountActivityStore?.update()
+    if let confirmationStore = confirmationStore {
+      confirmationStore.fetchDetails(for: confirmationStore.activeTransaction)
+    }
```

Comment:
Im a bit worried about having a single method control updating across the app when the currency changes. Wouldn't it be better to make the each store handle its own update based on currency change?